app/models/assignment.rb

```diff
@@ -1,11 +1,17 @@
 class Assignment < ApplicationRecord
   belongs_to :client
   accepts_nested_attributes_for :client
+
   belongs_to :volunteer
   accepts_nested_attributes_for :volunteer
+
   belongs_to :creator, class_name: 'User', foreign_key: 'creator_id'
 
-  validates :client_id, uniqueness: { scope: :volunteer_id, message: I18n.t('assignment_exists') }
+  has_one :import, as: :importable, dependent: :destroy
+  accepts_nested_attributes_for :import, allow_destroy: true
 
-  STATES = [:suggested, :active].freeze
+  STATES = [:suggested, :active, :finished, :archived].freeze
+
+  validates :client_id, uniqueness: { scope: :volunteer_id, message: I18n.t('assignment_exists') }
```

Comment:
We could/should also add a unique index on the DB level:

```ruby
add_index :assignments, [:client_id, :volunteer_id], unique: true
```

Reply:
Not my change actually, because I just moved the line. But never mind. I'll do it.

Reply:
Hold on, I need to take that back. I disagree. This is because that uniqueness validation is anyway not an implementation that can stay like this in the future. Because it expects a volunteer and a client to have only one assignment ever, ever and ever. I don't think that it is impossible that a volunteer has an assignment, and when it ends the volunteer has another assignment with the same client.

Reply:
It expects that duplicate assignment cannot exist. A volunteer can have many assignments based on our current implementation/validation.

Reply:
I'm not mentioning many assignments with different clients. Assignment A first, and then when assignment A is finished another Assignment with the same client again. That is well thinkable, as far as I assume.

Reply:
Plus, as the client has:

```ruby
has_one :assignment
```

Could it get more than one then anyway?

Reply:
Ah ok if you were just moving the line around we can leave it as it is.

It's just a general best practice to always enforce uniqueness on the DB level as well, because Rails can't guarantee it completely: https://robots.thoughtbot.com/the-perils-of-uniqueness-validations